src/pages/index.tsx

```diff
@@ -69,15 +70,17 @@ const PortalHomePage = ({
     }
   };
 
+  const router = useRouter();
+
   useEffect(() => {
     if (checkInResponse) {
       // In dev mode, this runs twice because of reactStrictMode in nextConfig.
       // This will only be run once in prod or deployment.
       processCheckInResponse(checkInResponse);
       // Clear the query params without re-triggering getServerSideProps.
-      window.history.replaceState(null, '', '/');
+      router.push(`${config.homeRoute}`, undefined, { shallow: true });
```

Comment:
Does this work? in my testing, after you check-in with the query params and `{ shallow: true }` re-renders w/o reloading props so the toast gets triggered infinitely